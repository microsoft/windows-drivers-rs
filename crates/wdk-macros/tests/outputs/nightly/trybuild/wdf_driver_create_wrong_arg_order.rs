../../../inputs/trybuild/wdf_driver_create_wrong_arg_order.rs