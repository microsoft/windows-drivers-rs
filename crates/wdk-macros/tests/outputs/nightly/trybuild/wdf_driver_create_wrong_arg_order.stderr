error[E0308]: arguments to this function are incorrect
  --> tests/outputs/nightly/trybuild/wdf_driver_create_wrong_arg_order.rs
   |
   | /         wdk_macros::call_unsafe_wdf_function_binding!(
   | |             WdfDriverCreate,
   | |             driver as PDRIVER_OBJECT,
   | |             registry_path,
   | |             // The order of the next two arguments is swapped!
   | |             &mut driver_config,
   | |             ------------------ expected `*mut _WDF_OBJECT_ATTRIBUTES`, found `&mut _WDF_DRIVER_CONFIG`
   | |             WDF_NO_OBJECT_ATTRIBUTES,
   | |             ------------------------ expected `*mut _WDF_DRIVER_CONFIG`, found `*mut _WDF_OBJECT_ATTRIBUTES`
   | |             driver_handle_output,
   | |         )
   | |_________^
   |
note: function defined here
  --> tests/outputs/nightly/trybuild/wdf_driver_create_wrong_arg_order.rs
   |
   | /         wdk_macros::call_unsafe_wdf_function_binding!(
   | |             WdfDriverCreate,
   | |             driver as PDRIVER_OBJECT,
   | |             registry_path,
...  |
   | |             driver_handle_output,
   | |         )
   | |_________^
   | |_________|
   | |_________|
   | |_________|
   | |_________|
   |
   = note: this error originates in the macro `wdk_macros::call_unsafe_wdf_function_binding` (in Nightly builds, run with -Z macro-backtrace for more info)
help: swap these arguments
   |
28 |         )(driver as PDRIVER_OBJECT, registry_path, WDF_NO_OBJECT_ATTRIBUTES, &mut driver_config, driver_handle_output)
   |          +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
